pkg/configfilearg/parser.go

```diff
@@ -101,14 +103,32 @@ func (p *Parser) findStart(args []string) ([]string, []string, bool) {
 		return []string{}, args, true
 	}
 
+	afterIndex := make(map[string]int)
+	re, err := regexp.Compile(`:\d`)
+	if err != nil {
+		return args, nil, false
+	}
+	// After keywords ending with ":<NUM>" will set + NUM of arguments as the split point.
+	// used for matching on subcommmands
+	for i, arg := range p.After {
+		if re.MatchString(arg) {
+			split := strings.Split(arg, ":")
+			p.After[i] = split[0]
+			afterIndex[split[0]], err = strconv.Atoi(split[1])
+			if err != nil {
+				return args, nil, false
+			}
+		}
+	}
+
 	for i, val := range args {
 		for _, test := range p.After {
 			if val == test {
-				return args[0 : i+1], args[i+1:], true
+				skip := afterIndex[test] + 1
```

Comment:
etcd-snapshot can be run with or without a subcommand; the default action if no subcommand is provided is to `save` I believe? We should only skip NUM arguments forward if the next argument isn't a flag (doesn't start with --).

Also, does this handle the potential for skipping past the end of the args list if insufficient arguments are provided?

Reply:
Now covered.